storm-core/src/jvm/org/apache/storm/serialization/KryoTupleDeserializer.java

```diff
@@ -50,7 +54,8 @@ public Tuple deserialize(byte[] ser) {
             List<Object> values = _kryo.deserializeFrom(_kryoInput);
             return new TupleImpl(_context, values, taskId, streamName, id);
         } catch(IOException e) {
-            throw new RuntimeException(e);
+            LOG.error("Failed to deserialize tuple.", e);
```

Comment:
Storm is designed to be a fail fast system, this fix appears to be doing the opposite of that.  If we cannot deserialize a tuple we need to make it very very clear to the user that something bad is happening.  Sadly in my experience even crashing the worker is not always enough, but in most cases it is.  Can we do a System.exit after logging the error instead of returning null?